torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -327,28 +227,162 @@ std::stringstream backport_v6_to_v5(std::stringstream& input_model_stream) {
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  Pickler data_pickle(
+      [&](const char* buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      nullptr,
+      &memoizedClassTypes);
+  data_pickle.protocol();
+  data_pickle.pushIValue(value);
+  data_pickle.stop();
+  size_t i = 0;
+  std::string prefix = archive_name + "/";
+
+  for (const auto& td : data_pickle.tensorData()) {
+    WriteableTensorData writable_td = getWriteableTensorData(td);
+    std::string fname = prefix + c10::to_string(i++);
+    writer.writeRecord(fname, writable_td.data(), writable_td.sizeInBytes());
+  }
+  std::string fname = archive_name + ".pkl";
+  writer.writeRecord(fname, data.data(), data.size());
+}
+
+std::stringstream backport_v5_to_v4(std::stringstream& input_model_stream) {
+  // 1) read from archive `bytecode` archive
+  PyTorchStreamReader reader(&input_model_stream);
+  std::vector<IValue> bytecode_values = get_bytecode_ivalues(reader);
+  std::vector<IValue> constants_values =
+      readArchive(kArchiveNameConstants, reader).toTuple()->elements();
+
+  // 2) Copy everything to new output, except some specific files and dirs
+  // (usually version, bytecode.pkl and bytecode folder are skipped)
+  std::unordered_set<std::string> excluded_files{
+      "constants.pkl",
+      "bytecode.pkl",
+      "version",
+  };
+
+  std::unordered_set<std::string> excluded_dirs{
+      "constants",
+      "bytecode",
+  };
+
+  std::stringstream ouput_model_stream;
+  auto writer_func = [&](const void* buf, size_t nbytes) -> size_t {
+    ouput_model_stream.write(static_cast<const char*>(buf), nbytes);
+    return !ouput_model_stream ? 0 : nbytes;
+  };
+
+  PyTorchStreamWriter writer(writer_func);
+
+  selective_copy(reader, writer, excluded_files, excluded_dirs);
+
+  // 3) write `bytecode` archive
+  // Update the bytecode version in bytecode.pkl
+  update_bytecode_version(bytecode_values, kBytecodeVersionV4);
+  // Construct the list of ivalues to a big tuple
+  auto bytecode_tuple = c10::ivalue::Tuple::create(std::move(bytecode_values));
+  // write `bytecode` archive
+  writeArchiveV4(writer, kArchiveNameBytecode, bytecode_tuple);
+  // write `constants` archive
+  auto constants_tuple =
+      c10::ivalue::Tuple::create(std::move(constants_values));
+  writeArchiveV4(writer, kArchiveNameConstants, constants_tuple);
+  return ouput_model_stream;
+}
+
+std::stringstream backport_v6_to_v5(std::stringstream& input_model_stream) {
+  bool emit_default_input_instructions = true;
+  bool enable_defaults_args_with_out_args = false;
+
+  std::stringstream ouput_model_stream = re_emit_model(
+      input_model_stream,
+      emit_default_input_instructions,
+      enable_defaults_args_with_out_args,
+      kBytecodeVersionV5 /*to_version*/);
 
   return ouput_model_stream;
 }
+
+std::stringstream backport_v7_to_v6(std::stringstream& input_model_stream) {
```

Comment:
Eg. documentation:

Backport function bytecode v7 that introduced support for operators with out arguments.
Previously, in v6, operators with out arguments forced the serialization of all arguments in the schema, even when optional arguments were not provided (as they had default values).
Currently, in v7, operators are aware of out arguments being present in the schema (always appended), allowing the serialization of only required arguments (as default values will be provided by the runtime).
The bump was needed because the v7 bytecode specifies less arguments for ops with out arguments in the schema, since the runtime code is now able to query whether an argument is of type "out" and insert the necessary default values in the right order in the interpreter stack (i.e. before the out arguments).
For example schema is: torch.add(x, h, alpha=1.0, out=x)
So, if the program defines: torch.add(x, h, out=x)
Previously, in v6, we serialized the bytecode to contain all 4 arguments.
Currently, in v7, we serialize the bytecode with only 3 arguments, since alpha is optional and has a default value that the runtime will push in the stack.
Thus, the backport is necessary such that the bytecode contains all the arguments as before.